Integrate GeniusAPI for the Lyrics command

Sebs5384 opened the pull request:
This will setup a new functionality which makes requests to the Genius API by using either prefix or slash commands

Sebs5384 committed:
Install lyrics library

We'll be using this for the new feature

Sebs5384 committed:
Installs dotenv

Using dotenv to process our access tokens

Sebs5384 committed:
Update lyrics

Setting up the main structure of the lyrics feature

Sebs5384 committed:
Setup configuration

Setting up the environment in pc

Sebs5384 committed:
Merge branch 'lyrics-dev' of https://github.com/Sebs5384/MissPiggyBot into lyrics-dev

Sebs5384 committed:
Gets the current song lyrics

This will replace the previous dummy implementation and add the actual song being played by the bot in order to get its lyrics, needs clean up.

Sebs5384 committed:
Update lyrics

This will clean up the code a little and return a message to the server if no song has been found

Sebs5384 committed:
Uninstalls dotenv

We will be using config.json to setup our tokens

Sebs5384 committed:
Update dependencies

Sebs5384 committed:
Update commands

This will add an array that will keep track of the userSongs as soon as he starts the bot by using /play

Sebs5384 committed:
Update lyrics command

This will add the required logic to find a more specific song lyrics, we keep track of the user and the queue songs to achieve this

Sebs5384 committed:
Updates lyrics

Moves the genius client build inside the run function to avoid double declaration inside slashRun

Sebs5384 committed:
Add buttons to emb

This will add the buttons structure, we'll figure out the pagination logic

Sebs5384 committed:
Update pagination

This will add up the currentPage and which will replace it the currentSong according to the interactions

Sebs5384 committed:
Add member to prefix and slash

Sebs5384 committed:
Setup pagination logic

This will handle the pagination logic that the user will use to browse through the list of songs lyrics returned by the API

Sebs5384 committed:
Update lyrics emb

This will add an emb that is returned when the song is too long to be displayed as a message.

Sebs5384 committed:
Removes console.log

arminherling commented on commands/lyrics.js:
```diff
@@ -1,39 +1,164 @@
-command.slashRun = async function slashRun(client, interaction)
-{
-    await interaction.followUp({ content: 'Lets ask Kermit!', ephemeral: true });
+command.aliases = ['.l', '.lyric']
```
You forgot to revert this change

Sebs5384 replied:
Reverted this change

arminherling commented on commands/lyrics.js:
```diff
@@ -1,39 +1,164 @@
+async function run(client, channel, send, member) {
+
+    const lyricsToken = client.lyricsToken
+    const geniusClient = new Genius.Client(lyricsToken)
```
It looks like client.lyricsToken is never set, maybe you forgot to commit that.
Maybe we can set client.genius = new Genius.Client instead of the token, that might be nicer.

Sebs5384 replied:
Added this change, it is now  client.genius rather than the constant geniusClient

arminherling commented on commands/lyrics.js:
```diff
@@ -1,39 +1,164 @@
+async function handlePagination(client, message, member, songList, send, currentPage){
+    const commandAuthorFilter = i => i.user.id === member.id
+
+    const collector = message.createMessageComponentCollector({ commandAuthorFilter, time: 300000 })
```
Could you add a comment that says how many second this is please?

Sebs5384 replied:
Added a comment and turned it into a constant instead of just a magic number

arminherling commented on commands/play.js:
```diff
@@ -1,5 +1,7 @@
 import { SlashCommandBuilder } from 'discord.js'
 
+export const userSongs = []
```
The bot can be connected to multiple voice channels and play different songs at the same time.
So instead of exporting this, we should try to put it on the queue because each vc has their own queue.

Sebs5384 replied:
Added a new key to the queue object and then used the previous exported array as its value to keep track of the userSongs, the playSong file will notice after the second song is being played this avoids an issue that comes down to playSong being executed first than play.js.

arminherling reviewed (commented):
Looks good so far.
There are some things like formatting but thats not important. 
As a note for the future, it might be nice to combine createLyricsButtons and createPlaylistButtons in playlist.js into another function to reduce code duplication.

Sebs5384 committed:
Update handlePagination

This will add a new constant that holds the collector timeout for the buttons before the user can't interact with them no more, it also indents the new Emb correctly.

Sebs5384 committed:
Update userSongs queue

This will add a new key to the queue object for each queue in each voice channel the bot is currently playing on, instead of our previous implementation which was dependent of exporting the array itself

Sebs5384 committed:
Removes dots on command aliases

Turning this back to what the real bot have

Sebs5384 committed:
Update playSong condition

This will update the condition to shift the userSong queue

arminherling commented:
LGTM
